shared/src/main/scala/mlscript/TyperDatatypes.scala

```diff
@@ -535,7 +535,7 @@ abstract class TyperDatatypes extends TyperHelpers { Typer: Typer =>
       _assignedTo = value
     }
 
-    var tsc: MutMap[TupleSetConstraints, Int] = MutMap.empty
+    var tsc: LinkedHashMap[TupleSetConstraints, Int] = LinkedHashMap.empty
```

Comment:
BTW why is this a `var`?

Reply:
It used to be an Option so `var` is used. I have changed it to `val`.